torch/csrc/autograd/profiler_kineto.cpp

```diff
@@ -497,6 +497,7 @@ void pushProfilingCallbacks(const std::unordered_set<at::RecordScope>& scopes) {
           onFunctionEnter<use_global_callback>,
           onFunctionExit<use_global_callback>)
           .needsInputs(registration_state_ptr->config().report_input_shapes)
+          .needsIds(true)
```

Comment:
@davidberard98 brought up a good point about contention on a global atomic id. I think we should try to avoid contention on global variables, we switched the profiler to use thread local states to avoid such contention. Will this make the profiler go back to being delayed by global contention?